packages/bus-rdf-resolve-quad-pattern/lib/ActorRdfResolveQuadPattern.ts

```diff
@@ -27,6 +27,15 @@ export function getDataSourceType(dataSource: IDataSource): string | undefined {
 export function getDataSourceValue(dataSource: IDataSource): string | RDF.Source {
   return isDataSourceRawType(dataSource) ? dataSource : dataSource.value;
 }
+export function getExtraSourceParams(dataSource: IDataSource): any {
```

Comment:
(make sure to read the next comment first)

I would rename this to `getDataSourceContext`, give it a second param (`originalContext: ActionContext`), and return a new ActionContext that is the merge of the original context and the datasource-specific context. If no datasource-specific context exist, you can just return the original context.